### PR TITLE
[FIX] mail, base_automation: fix composer tour without demo

### PR DESCRIPTION
**Pull request by seb-odoo**
runbot-26628

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@robodoo r-

You should create a test partner for the tour, and replace mentions of "Marc Demo" in the tour itself.

**Comment by tde-banana-odoo** [comment]
Indeed, it allows to stay completely independent. Mail already has a lot of test data, it is sufficient for most (all ?) tests. Moreover you can after the tour check that partner is in the recipients for example, making the test more explicit.

**Comment by seb-odoo** [comment]
updated

File: addons/base_automation/tests/test_mail_composer.py [hunk 1]
```diff
@@ -34,11 +34,12 @@ def test_mail_composer_test_tour(self):
             'model_id': self.env.ref('mail.model_mail_compose_message').id,
         })
         automation.write({'action_server_ids': [(4, server_action.id)]})
-
+        user = self.env["res.users"].create({"name": "Not A Demo User", "login": "nadu"})
```
Comment (tde-banana-odoo) [inline addons/base_automation/tests/test_mail_composer.py:37]: But do we need a user ? I mean, purpose is to test recipients, so it seems more logical to use partners ?

File: addons/base_automation/tests/test_mail_composer.py [hunk 2]
```diff
@@ -34,11 +34,12 @@ def test_mail_composer_test_tour(self):
...
         self.start_tour(
             f"/web#id={self.user_employee.partner_id.id}&model=res.partner",
             "mail/static/tests/tours/mail_composer_test_tour.js",
             login=self.user_employee.login
         )
-
+        message = self.env["mail.message"].search([], order="id desc")
```
Comment (tde-banana-odoo) [inline addons/base_automation/tests/test_mail_composer.py:43]: Can't we use existing mocks and be sure we produced a message during the tour, and check its values ?

**Review by tde-banana-odoo — COMMENTED** [review]
Thanks, some additional comments on my side :)

File: addons/base_automation/tests/test_mail_composer.py (hunk 2 above)
Comment (seb-odoo) [reply addons/base_automation/tests/test_mail_composer.py:43]: I don't know about them but that looks like a good idea.

**Comment by tde-banana-odoo** [comment]
@robodoo r+

Thanks for update :) did not know there was a difference for users and partners. Anyway good enough for stable :)

**Pull request closed without merging**